publication-commons/src/main/java/no/unit/nva/publication/model/FilesApprovalEntry.java

```diff
@@ -91,8 +93,8 @@ public Set<File> getFilesForApproval() {
         return nonNull(filesForApproval) ? filesForApproval : Set.of();
     }
 
-    public void setFilesForApproval(Set<File> filesForApproval) {
-        this.filesForApproval = filesForApproval;
+    public void setFilesForApproval(Collection<File> filesForApproval) {
```

Comment:
Because we have done this and there is only a zero-arg constructor (implicit?).

I'd actually consider initialising the Set in the field and just call `addAll` on this in the setter to avoid nulls.

And replace:
```
    public Set<File> getFilesForApproval() {
        return nonNull(filesForApproval) ? filesForApproval : Set.of();
    }
```

with

```
    public Set<File> getFilesForApproval() {
        return new HashSet<>(filesForApproval);
    }
``` 

This will then prevent mutating the object accidentally in subsequent ops.

This style makes immutable objects, but it might require a helper-constructor for people who want to mutate things (if they do this) so they can chain this object to another i.e. this.copy().setFields(old + new)

Reply:
If setFiles...() will use addAll() method, we should change whole implementation of how these collections are used and mutated in production code. Cause instead of overwriting collection as we do now, we should then both remove items and add items to it, when file is being added to or removed from publication.